test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -186,6 +189,10 @@ public void setBlockOnWriteIndexFile(boolean blocked) {
         blockOnWriteIndexFile = blocked;
     }
 
+    public void setBlockAndFailOnWriteIndexFile(boolean blocked) {
```

Comment:
Had to add another method here for the existing uses of blocking on index files that would in fact always fail (unlike the name suggests) so I could make the existing method only block but not error out in the min threads test.